src/game/common/thing/thingfactory.cpp

```diff
@@ -13,15 +13,167 @@
  *            LICENSE
  */
 #include "thingfactory.h"
+#include "gameclient.h"
+#include <cstring>
+
 #ifndef GAME_DLL
 ThingFactory *g_theThingFactory = nullptr;
 #endif
 
-ThingTemplate *ThingFactory::Find_Template(Utf8String name)
+ThingFactory::ThingFactory() : m_firstTemplate(nullptr), m_nextTemplateID(0) {}
+
+ThingFactory::~ThingFactory()
+{
+    Free_Database();
+}
+
+ThingTemplate *ThingFactory::New_Template(const Utf8String &name)
+{
+    ThingTemplate *new_template = NEW_POOL_OBJ(ThingTemplate);
+    const ThingTemplate *default_template = Find_Template("DefaultThingTemplate", false);
+
+    if (default_template) {
+        *new_template = *default_template;
+        new_template->Set_Copied_From_Default();
+    }
+
+    new_template->Friend_Set_Template_ID(m_nextTemplateID);
+    m_nextTemplateID++;
+
+    if (!m_nextTemplateID) {
+        captainslog_debug("m_nextTemplateID wrapped to zero");

```

Comment:
This was a captainselog_dbgassert rather than  just a dbg message.

Reply:
Fixed.